src/XSAlgo/XSAlgo_ShapeProcessor.cxx

```diff
@@ -0,0 +1,548 @@
+// Created on: 2024-12-13
+// Created by: data exchange team
+// Copyright (c) 2000-2014 OPEN CASCADE SAS
+//
+// This file is part of Open CASCADE Technology software library.
+//
+// This library is free software; you can redistribute it and/or modify it under
+// the terms of the GNU Lesser General Public License version 2.1 as published
+// by the Free Software Foundation, with special exception defined in the file
+// OCCT_LGPL_EXCEPTION.txt. Consult the file LICENSE_LGPL_21.txt included in OCCT
+// distribution for complete text of the license and disclaimer of any warranty.
+//
+// Alternatively, this file may be used under the terms of Open CASCADE
+// commercial license or contractual agreement.
+
+#include "XSAlgo_ShapeProcessor.hxx"
+
+#include <BRep_Builder.hxx>
+#include <BRep_Tool.hxx>
+#include <BRepBuilderAPI_MakeEdge.hxx>
+#include <BRepTools_ReShape.hxx>
+#include <Geom_Curve.hxx>
+#include <Interface_Static.hxx>
+#include <Message_ListOfMsg.hxx>
+#include <Resource_Manager.hxx>
+#include <ShapeAnalysis_Edge.hxx>
+#include <ShapeBuild_Edge.hxx>
+#include <ShapeExtend_MsgRegistrator.hxx>
+#include <ShapeFix_Edge.hxx>
+#include <ShapeProcess_ShapeContext.hxx>
+#include <TopExp_Explorer.hxx>
+#include <TopoDS.hxx>
+#include <Transfer_FinderProcess.hxx>
+#include <Transfer_TransientListBinder.hxx>
+#include <Transfer_TransientProcess.hxx>
+#include <TransferBRep.hxx>
+#include <TransferBRep_ShapeBinder.hxx>
+#include <TransferBRep_ShapeMapper.hxx>
+#include <UnitsMethods.hxx>
+
+//=======================================================================
+//function : XSAlgo_ShapeProcessor
+//purpose  : 
+//=======================================================================
+XSAlgo_ShapeProcessor::XSAlgo_ShapeProcessor(const ParameterMapPtr  theParameters,
+                                             const Standard_Real    theTolerance,
+                                             const Standard_Real    theMaxTolerance,
+                                             const TopAbs_ShapeEnum theDetalizationLevel,
+                                             const Standard_Boolean theNonManifold)
+  : myParameters(theParameters),
```

Comment:
In context if we have a MAP. I expect that not-initialized values that are not exists in map be filled up by default. Is it sounds reasonable?